src/precompile.jl

```diff
@@ -3,5 +3,41 @@ function _precompile_()
     precompile(Tuple{typeof(Parsers.parse), Type{Int64}, String})
     precompile(Tuple{typeof(Parsers.parse), Type{Float64}, String})
     precompile(Tuple{typeof(Parsers.parse), Type{Date}, String})
+
+    options = Parsers.Options()
+    pos = 1
+    val = "a"
+    len = length(val)
+    for T in (Char, String), buf in (codeunits(val), Vector(codeunits(val)))
```

Comment:
should also InlineStrings.jl precompilation be included here?

Reply:
I wouldn't think they could be included here since they're in a separate package (InlineStrings.jl); but we could do similar precompilation over there.

Reply:
The dependency goes the other way. Should we add a similar precompile block over there?

Reply:
I don't actually get much improvement with precompilation in InlineStrings.jl. Precompiling for all types only takes half a second there.

Reply:
What do you mean by "Precompiling for all types only takes half a second there"? Precompiling all inlinestring types? I wouldn't be too surprised by that since the heavy lifting would be handled by precompiling `String` in the Parsers.jl package.

Reply:
Yes, all the InlineString types. Without `String` in Parsers.jl precompilation adds under a second to InlineString precompilation.

It seems there is something about the numerical methods in Parsers.jl that takes a lot longer.